sql/2024/cookies/prevalence_type_attributes_per_rank.sql

```diff
@@ -0,0 +1,26 @@
+-- Prevalence of cookies type and attributes per rank grouping
+-- Before running query: edit table name
+
+SELECT
+  rank_grouping,
+  SUM(IF(firstPartyCookie = TRUE, 1, 0)) / COUNT(0) AS firstParty,
+  SUM(IF(firstPartyCookie = FALSE, 1, 0)) / COUNT(0) AS thirdParty,
+  SUM(IF(httpOnly = 'true', 1, 0)) / COUNT(0) AS httpOnly,
+  SUM(IF(secure = 'true', 1, 0)) / COUNT(0) AS secure,
+  SUM(IF(session = 'true', 1, 0)) / COUNT(0) AS session,
+  SUM(IF(sameParty = 'true', 1, 0)) / COUNT(0) AS sameParty,
+  SUM(IF(sameSite = 'Lax', 1, 0)) / COUNT(0) AS sameSiteLax,
+  SUM(IF(sameSite = 'None', 1, 0)) / COUNT(0) AS sameSiteNone,
+  SUM(IF(sameSite = 'Strict', 1, 0)) / COUNT(0) AS sameSiteStrict,
+  SUM(IF(sameSite IS NULL, 1, 0)) / COUNT(0) AS sameSiteNull,
+  SUM(IF(partitionKey IS NOT NULL, 1, 0)) / COUNT(0) AS partitionKey,
+  SUM(IF(partitionKeyOpaque IS NOT NULL, 1, 0)) / COUNT(0) AS partitionKeyOpaque
+FROM `httparchive.almanac.DATE_CLIENT_RANK_cookies`,
+  UNNEST([1000, 5000, 10000, 50000, 100000, 500000, 1000000]) AS rank_grouping
```

Comment:
We normally don't include the mid-steps in Web Almanac and just report in the "10s":

```suggestion
  UNNEST([1000, 10000, 100000, 1000000, 10000000, 100000000]) AS rank_grouping
```

Also added two more groupings just in case this is expanded above.

Reply:
Yeah. More confusing than anything else so let’s be consistent.